core/src/main/java/org/kohsuke/stapler/annotations/StaplerPath.java

```diff
@@ -0,0 +1,372 @@
+                if (implicit != null) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public static boolean isDynamic(Function method) {
+            return isMethodDynamic(method.getAnnotations());
+        }
+
+        public static boolean isDynamic(Method method) {
+            return Modifier.isPublic(method.getModifiers()) && isMethodDynamic(method.getAnnotations());
+        }
+
+        public static boolean isDynamic(MethodRef method) {
+            return method.isRoutable() && isMethodDynamic(method.getAnnotations());
+        }
+
+        private static boolean isMethodDynamic(Annotation[] annotations) {
+            for (Annotation a : annotations) {
+                if (a instanceof StaplerPaths) {
+                    for (StaplerPath p : ((StaplerPaths) a).value()) {
+                        if (DYNAMIC.equals(p.value())) {
+                            return true;
+                        }
+                    }
+                } else if (a instanceof StaplerPath) {
+                    StaplerPath p = (StaplerPath) a;
+                    if (DYNAMIC.equals(p.value())) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        public static Iterable<String> getPaths(Function method) {
+            return getMethodPaths(method.getName(), method.getAnnotations());
+        }
+
+        public static Iterable<String> getPaths(Method method) {
+            if (!Modifier.isPublic(method.getModifiers())) {
+                return Collections.emptyList();
+            }
+            return getMethodPaths(method.getName(), method.getAnnotations());
+        }
+
+        public static Iterable<String> getPaths(MethodRef method) {
+            if (!method.isRoutable()) {
+                return Collections.emptyList();
+            }
+            return getMethodPaths(method.getName(), method.getAnnotations());
+        }
+
+        private static Iterable<String> getMethodPaths(String methodName, Annotation[] annotations) {
+            Set<String> names = new TreeSet<>();
+            Set<String> prefixes = null; // lazy init to avoid an allocation
+            boolean hasPathAnnotation = false;
+            boolean hasImplicitAnnotation = false;
+            boolean hasInferredName = false;
+            for (Annotation a: annotations) {
+                if (a instanceof StaplerPaths) {
+                    hasPathAnnotation = true;
+                    StaplerPath[] paths = ((StaplerPaths) a).value();
+                    if (paths.length == 0) {
+                        hasInferredName = true;
+                    } else {
+                        for (StaplerPath p: paths) {
+                            switch (p.value()) {
+                                case DYNAMIC:
+                                    continue;
```

Comment:
it's weird to use a `continue` there that will jump back to the `for loop` 10 lines above but the next `break`s are relative to the `switch`. Not intuitive to read.